### PR TITLE
Node Explorer: Use the MagicDNS name when appropriate

### PR DESCRIPTION
**Pull request by tylersmalley**
If the MagicDNS is enabled, and the tailnet name is an email address (includes an @), use the MagicDNSName

image

### CODE REVIEW

File: src/utils/string.ts [hunk 1]
```diff
@@ -1,8 +1,4 @@
-export function trimSuffix(str: string | undefined, suffix: string) {
-  if (!str) {
-    return;
-  }
-
+export function trimSuffix(str: string, suffix: string): string {
```
Comment (marwan-at-work) [inline src/utils/string.ts:1]: I think this breaks https://github.com/tailscale-dev/vscode-tailscale/blob/6eb760448be2339fd456bf75a21e608c7bc48750/src/tailscale/cli.ts#L442

Or from what I can tell, in Go `Self` might actually be nil but we're not specifying so in the TypeScript type decl: https://github.com/tailscale-dev/vscode-tailscale/blob/6eb760448be2339fd456bf75a21e608c7bc48750/src/types.ts#L52

Comment (tylersmalley) [reply src/utils/string.ts:1]: Reverted the change and accounted for it returning undefined.

Comment (tylersmalley) [reply src/utils/string.ts:1]: Well, now waiting for it to show up on Github...

Comment (tylersmalley) [reply src/utils/string.ts]: Ok, closing and re-opening the PR fixed it. Github lost that this PR was associated with the branch somehow as seen in the branch list.

**Pull request closed without merging**